state-synchronizer/src/coordinator.rs

```diff
@@ -731,6 +731,12 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
             None
         };
         if let Some(li) = end_of_epoch_li.as_ref() {
+            ensure!(
+                li.ledger_info().version() >= request.known_version,
+                "end-of-epoch LI version {} < request waypoint version {}",
+                li.ledger_info().version(),
+                request.known_version,
```

Comment:
should be able to use local_state.trusted_epoch to verify it?

Reply:
this is on the end of sending the chunk response, not processing it. This checks that request's current_epoch and request's known_version match up and doesn't cause overflow below.

Reply:
I'm a little concerned about adding extra checks at this point in time. Is it that bad if we just ... get rid of this? or make it a warning instead of a Result?

Reply:
if this check isn't included, we will encounter subtraction overflow right in the line below (line 741)